pkg/query-service/app/opamp/opamp_server.go

```diff
@@ -0,0 +1,132 @@
+package opamp
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/knadh/koanf/parsers/yaml"
+	"github.com/open-telemetry/opamp-go/protobufs"
+	"github.com/open-telemetry/opamp-go/server"
+	"github.com/open-telemetry/opamp-go/server/types"
+	"go.opentelemetry.io/collector/confmap"
+	model "go.signoz.io/signoz/pkg/query-service/app/opamp/model"
+	"go.uber.org/zap"
+)
+
+type Server struct {
```

Comment:
Can we instead build an interface and provide implementation in form of opamp.  This will allow moving away from opAmp (if we have to) to home-grown or other approaches. 

I think we could also keep opAmp in integrations folder as it is mainly used for communication and integrates with other system.

Reply:
I was thinking of not building out abstractions prematurely  when things are early/uncertain. Since what we are already doing is an implementation of the opamp, adding more layers or thinking of home-grown solutions is probably too futuristic. I believe we can revisit this as we build and learn more.